src/utils/setupSubjectChannels.ts

```diff
@@ -0,0 +1,59 @@
+import { Guild, MessageActionRow, MessageSelectMenu } from 'discord.js';
+import { channelSelect } from '../selects';
+import { Config } from './config';
+
+function capitalize(str: string) {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+export async function setupSubjectChannels(guild: Guild) {
+    let channelGroupIndex = 0;
+    // Apparently this deletes all elements with index > 0
+    channelSelect.data.length = 0;
+    for (const channelGroupId of Config.Instance.Properties
+        .SubjectChannelGroupIDs) {
+        const channelGroup = guild.channels.cache.get(channelGroupId);
+        const categoryChannels = guild.channels.cache.filter(
+            (x) => x.parentId === channelGroupId
+        );
+        if (!channelGroup || !categoryChannels || categoryChannels.size < 1) {
+            // we don't have a channel group or it's empty
+            console.log(
+                `Channel Group ID ${channelGroupId} is either not a group, or is empty`
+            );
+            continue;
+        }
+        channelSelect.data.push(
+            new MessageActionRow<MessageSelectMenu>().addComponents(
+                new MessageSelectMenu()
+                    .setCustomId('channelSelect-' + channelGroupId)
+                    .setPlaceholder(channelGroup.name)
+                    .setMinValues(1)
+            )
+        );
+        if (categoryChannels) {
+            for (const subjectChannel of categoryChannels) {
+                let subjectName = subjectChannel[1].name;
+                const nameParts = subjectName.split('-');
+                subjectName = `${nameParts[0].toUpperCase()} ${nameParts
+                    .splice(1)
+                    .map(capitalize)
+                    .join(' ')}`;
+                channelSelect.data[channelGroupIndex].components[0].addOptions([
+                    {
+                        label: subjectName,
+                        value: subjectChannel[1].id,
+                    },
+                ]);
+            }
+            channelSelect.data[channelGroupIndex].components[0].setMaxValues(
+                Math.min(categoryChannels.size, 25) // 25 is Discord API max
```

Comment:
Maybe error message would be nice somewhere if max is exceeded.

Reply:
This only means that a user is able to select at most `categoryChannels.size` or 25 values at the same time, if we exceed 25, it's not wrong, just a small user inconvenience.